rules/manager_test.go

```diff
@@ -326,6 +326,21 @@ func TestForStateAddSamples(t *testing.T) {
 	}
 }
 
+// sortAlerts sorts `[]*Alert` w.r.t. the Labels.
+// This is cleaner for small slice than adding Len(), Less(), Swap() for []*Alert.
+func sortAlerts(items []*Alert) {
+	var n = len(items)
+	for i := 1; i < n; i++ {
+		j := i
+		for j > 0 {
+			if items[j-1].Labels.Hash() > items[j].Labels.Hash() {
```

Comment:
We can use `sort.Slice` here no?

Also, use https://godoc.org/github.com/prometheus/common/model#LabelSet.Before for comparing?

Reply:
Thanks for pointing me to `sort.Slice`. 

And `alert.Labels` is a slice of labels, whereas `LabelSet` is a map. Hence we will have unnecessary conversion to make?

Reply:
https://github.com/prometheus/prometheus/blob/134ba8e03ecd097d4f7a240c4e09a27d847ff389/pkg/labels/labels.go#L223-L225

This should be better :)

Reply:
Ah, I missed that while looking at the API. Updated now.